feat: add direnv gitignore, migrate flake to use pnpm, and create automatic hash update script (added to github action `made bot sign the cla`)

connerohnesorge opened the pull request:
This pull request updates the `flake.nix` file to improve its structure and maintainability. The most important changes include reformatting the `outputs` attribute for better readability and updating the `codex-cli` package definition to use a local path and a new hash for `npmDepsHash`.

### Structural improvements:
* Reformatted the `outputs` attribute to use a block-style layout, making it more readable and consistent with common Nix practices.

### Package definition updates:
* Updated the `codex-cli` package definition to use a local path (`./codex-cli`) for the `src` attribute instead of a relative reference (`self + "/codex-cli"`).
* Changed the `npmDepsHash` for `codex-cli` to a new value (`sha256-UkvkaM7tvVlio0st8UA45x8wQ+q423BTzshsmdmmO2o=`), reflecting updated dependencies.

### Autonomous Update for the future:
* Created a github action step and bash script for updating the npmHash in the flake.nix if the package-lock.json changes in ./codex-cli

connerohnesorge committed:
add .direnv and CODEX.md to .gitignore

connerohnesorge committed:
update hash in flake.nix

conneroisu commented:
I have read the CLA Document and I hereby sign the CLA

connerohnesorge committed:
added update-nix-hash script to automatically update nix hash in the flake.nix

connerohnesorge committed:
Merge branch 'main' into main

connerohnesorge committed:
fix update script after testing more in act

tibo-openai commented:
Thanks! One CODEX.md, why would we want to add it to .gitignore? It should be pretty universal, so what is the thinking here?

conneroisu commented:
Ope. thought the idea was to add more specific context to a goal in CODEX.md. I'll remove it from gitignore

connerohnesorge committed:
Merge branch 'main' into main

connerohnesorge committed:
Update .gitignore

remove CODEX.md from .gitignore

connerohnesorge committed:
Merge branch 'main' into main

tibo-openai commented:
This change is based on the package-lock.json, however we are using pnpm since https://github.com/openai/codex/pull/287. Could you update the logic to reflect?

conneroisu commented:
Yes I will update the logic to reflect. Thanks for switching to pnpm I like it alot more!

connerohnesorge committed:
Merge branch 'main' into main

connerohnesorge committed:
Merge branch 'main' into main

jonchurch commented on .github/workflows/ci.yml:
```diff
@@ -65,5 +65,56 @@ jobs:
       - name: Build
         run: pnpm run build
+        
+  update-nix-hash:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+    permissions:
+      contents: write  # Needed for push access
+      pull-requests: write  # Needed for PR creation
```
Just flagging this for awareness, this job introduces write access from CI for the first time (excluding the CLA Assistant, which already operates with elevated perms via `pull_request_target`).

It’s scoped well (only runs on push to main) so it won’t run on PRs or forks. That said, it does slightly shift the trust boundary, merging into main will now trigger a job that can push branches and open PRs back to the repo.

No action needed here, but it might be worth making sure folks reviewing automation changes are aware of that shift. If main isn’t already protected, enabling “require PRs before merging” and disallowing force-pushes/deletion would be good hardening steps for this kind of setup.

jonchurch replied:
since this job is the only one in ci.yml with write permissions, it might be worth considering moving it to a separate workflow file (e.g. `update-nix-hash.yml`). That could make it easier to see at a glance which workflows mutate the repo and which are strictly read-only CI.

Also worth noting, the script `update-nix-hash.sh` will be run with elevated permissions and therefore also needs special attention during review to ensure it is not going to smuggle a secret, accidentally commit something, and that it’s not vulnerable to input injection (e.g. using PR metadata like branch names in shell commands without quoting).

connerohnesorge committed:
working pnpm package building and automatic hash update

connerohnesorge committed:
working and tested fully automated update of nix hash

connerohnesorge committed:
get pnpm globally in flake

connerohnesorge committed:
Merge branch 'main' into main

conneroisu commented:
I updated the flake and hash updater to use pnpm, let me know if the github action step to update the flake hash should be it's own action.

connerohnesorge committed:
Merge pull request #1 from conneroisu/conneroisu/main

Conneroisu/main

connerohnesorge committed:
Merge branch 'main' into main

connerohnesorge committed:
Merge branch 'main' into main

connerohnesorge committed:
Merge branch 'main' into main

conneroisu commented:
I updated the flake to use pnpm. Please let me know if you want any other changes

connerohnesorge committed:
Merge branch 'main' into main

peteringram0 commented:
Related to: https://github.com/openai/codex/issues/515

connerohnesorge committed:
Merge branch 'main' into main

actions-user committed:
chore: Update pnpm deps hash in flake.nix

This automated commit updates the pnpm deps hash hash in flake.nix to match
the latest hash of the workspace's pnpm-lock.yaml file. This ensures that
Nix's store is consistent with the latest dependencies.

connerohnesorge committed:
Merge branch 'main' into main

connerohnesorge committed:
add nix workflow and escape if nix-hash gets "" or null

connerohnesorge committed:
removed nix hash updater from ci and put in a separate script/workflow

connerohnesorge committed:
Merge branch 'main' into conneroisu/main

connerohnesorge committed:
Merge pull request #2 from conneroisu/conneroisu/main

Conneroisu/main

frobware commented:
I see: 

```
% gh pr checkout 366
remote: Enumerating objects: 125, done.
remote: Counting objects: 100% (77/77), done.
remote: Compressing objects: 100% (25/25), done.
remote: Total 125 (delta 59), reused 59 (delta 52), pack-reused 48 (from 2)
Receiving objects: 100% (125/125), 32.68 KiB | 32.68 MiB/s, done.
Resolving deltas: 100% (73/73), completed with 24 local objects.
From https://github.com/openai/codex
 * [new ref]         refs/pull/366/head -> conneroisu/main
Switched to branch 'conneroisu/main'
% nix build .
warning: found empty hash, assuming 'sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA='
these 2 derivations will be built:
  /nix/store/hbkh142xr9gr8bvqwlj20z0cad2mf1bm--openai-codex-pnpm-deps.drv
  /nix/store/dxywbjmg0yrm194qs1ymhwadl3r2711j--openai-codex-0.1.2504221401.drv
these 18 paths will be fetched (6.45 MiB download, 24.18 MiB unpacked):
  /nix/store/8v6k283dpbc0qkdq81nb6mrxrgcb10i1-gcc-wrapper-14-20241116
  /nix/store/ld8ys81ch5fj94j0jrwbkgwwjxmbwdxh-jq-1.7.1-dev
  /nix/store/a28v2wclwmaymykakvbzfgzsxrqp4jij-libyaml-0.2.5
  /nix/store/i11sfnd6bx8xfrhijdja7lzqbw011dqm-moreutils-0.70
  /nix/store/9zxb5ln0awpc13s0gv1vz7f28wyvn2x1-nss-cacert-3.108
  /nix/store/fwswaix66sb58i6a05l117z41rjqbimr-perl-5.40.0-env
  /nix/store/ak6014vr8h3xf38vbrwjf26391fxn04a-perl-5.40.0-man
  /nix/store/znm51m3i16lc9af3ihsh785cln4yk09f-perl5.40.0-IO-Tty-1.17
  /nix/store/pcl2y9rsssr9bgb1563qmy6j03381w70-perl5.40.0-IPC-Run-20231003.0
  /nix/store/2w6lvqpfmiq2dqfjxyhlmwvgjj3qdq40-perl5.40.0-Time-Duration-1.21
  /nix/store/kclz9i3vq2fz8nnn947l773a95fj77z3-pnpm-10.8.0
  /nix/store/k2yvi58cqb5x9z3xrvry1msi3dfqvmc8-pnpm-config-hook
  /nix/store/dds9yi6lc5zh41n5k4qnijq4xjk7069v-python3.12-argcomplete-3.5.3
  /nix/store/smm6sag2w822b34mxs15ff3bp5x9fqqj-python3.12-pyyaml-6.0.2
  /nix/store/z05br9nvkhrw28p7wak6biz8a75silmm-python3.12-tomlkit-0.13.2
  /nix/store/gj0yy0gyv9y4m31f2b7x6lzr274xbyf6-python3.12-xmltodict-0.14.2
  /nix/store/id5jpgm9qi62am73zqc40lp124x3as35-python3.12-yq-3.4.3
  /nix/store/9r1nl9ksiyszy4qzzg6y2gcdkca0xmhy-stdenv-linux
copying path '/nix/store/8v6k283dpbc0qkdq81nb6mrxrgcb10i1-gcc-wrapper-14-20241116' from 'https://cache.nixos.org'...
copying path '/nix/store/9zxb5ln0awpc13s0gv1vz7f28wyvn2x1-nss-cacert-3.108' from 'https://cache.nixos.org'...
copying path '/nix/store/kclz9i3vq2fz8nnn947l773a95fj77z3-pnpm-10.8.0' from 'https://cache.nixos.org'...
copying path '/nix/store/ld8ys81ch5fj94j0jrwbkgwwjxmbwdxh-jq-1.7.1-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/ak6014vr8h3xf38vbrwjf26391fxn04a-perl-5.40.0-man' from 'https://cache.nixos.org'...
copying path '/nix/store/2w6lvqpfmiq2dqfjxyhlmwvgjj3qdq40-perl5.40.0-Time-Duration-1.21' from 'https://cache.nixos.org'...
copying path '/nix/store/dds9yi6lc5zh41n5k4qnijq4xjk7069v-python3.12-argcomplete-3.5.3' from 'https://cache.nixos.org'...
copying path '/nix/store/z05br9nvkhrw28p7wak6biz8a75silmm-python3.12-tomlkit-0.13.2' from 'https://cache.nixos.org'...
copying path '/nix/store/gj0yy0gyv9y4m31f2b7x6lzr274xbyf6-python3.12-xmltodict-0.14.2' from 'https://cache.nixos.org'...
copying path '/nix/store/a28v2wclwmaymykakvbzfgzsxrqp4jij-libyaml-0.2.5' from 'https://cache.nixos.org'...
copying path '/nix/store/znm51m3i16lc9af3ihsh785cln4yk09f-perl5.40.0-IO-Tty-1.17' from 'https://cache.nixos.org'...
copying path '/nix/store/pcl2y9rsssr9bgb1563qmy6j03381w70-perl5.40.0-IPC-Run-20231003.0' from 'https://cache.nixos.org'...
copying path '/nix/store/smm6sag2w822b34mxs15ff3bp5x9fqqj-python3.12-pyyaml-6.0.2' from 'https://cache.nixos.org'...
copying path '/nix/store/9r1nl9ksiyszy4qzzg6y2gcdkca0xmhy-stdenv-linux' from 'https://cache.nixos.org'...
copying path '/nix/store/id5jpgm9qi62am73zqc40lp124x3as35-python3.12-yq-3.4.3' from 'https://cache.nixos.org'...
copying path '/nix/store/fwswaix66sb58i6a05l117z41rjqbimr-perl-5.40.0-env' from 'https://cache.nixos.org'...
copying path '/nix/store/i11sfnd6bx8xfrhijdja7lzqbw011dqm-moreutils-0.70' from 'https://cache.nixos.org'...
copying path '/nix/store/k2yvi58cqb5x9z3xrvry1msi3dfqvmc8-pnpm-config-hook' from 'https://cache.nixos.org'...
building '/nix/store/hbkh142xr9gr8bvqwlj20z0cad2mf1bm--openai-codex-pnpm-deps.drv'...
error: hash mismatch in fixed-output derivation '/nix/store/hbkh142xr9gr8bvqwlj20z0cad2mf1bm--openai-codex-pnpm-deps.drv':
         specified: sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=
            got:    sha256-ZXX9f/MsqMN8fxTclowDdD8+gSOOhI0i+EduTJ5eM34=
error: 1 dependencies of derivation '/nix/store/dxywbjmg0yrm194qs1ymhwadl3r2711j--openai-codex-0.1.2504221401.drv' failed to build
```

connerohnesorge committed:
fix git merge flake output

connerohnesorge committed:
Merge pull request #3 from conneroisu/conneroisu/main

fix git merge flake output

connerohnesorge committed:
move towards python scripted updates of the hash

connerohnesorge committed:
Merge pull request #4 from conneroisu/conneroisu/main

move towards python scripted updates of the hash

connerohnesorge committed:
Merge branch 'main' into main

conneroisu commented:
Should be much better now. Switched from updating using bash to using python in an flake derived package/bin.

actions-user committed:
chore: Update pnpm deps hash in flake.nix
This automated commit updates the pnpm deps hash hash in flake.nix to match
the latest hash of the workspace's pnpm-lock.yaml file. This ensures that
Nix's store is consistent with the latest dependencies.

conneroisu commented:
It worked for me on a fresh clone and using act for the updating of the github action.
Right now to manually update the hash (rather than just editing the sha string manually) you can run `nix run .\#update-nix-hash`

connerohnesorge committed:
Merge branch 'main' into main

connerohnesorge committed:
have the github action bot sign the CLA

conneroisu commented:
Just updated the bot to sign the CLA since the CLA is failing

connerohnesorge committed:
Merge branch 'main' into main

connerohnesorge committed:
Merge branch 'main' into main

connerohnesorge committed:
update name fo workflow to update the pnpm hash in the flake.nix file

connerohnesorge committed:
fix typo in name for updating nix npm hash to be pnpm hash since the updates use pnpm

connerohnesorge committed:
simpler check if changes are needed

connerohnesorge committed:
Merge branch 'main' into main

peteringram0 commented:
@conneroisu Sorry im still new to nix/flakes so if this question is beside the point of this PR then please ignore it but if if i run 

```bash
~ ❯ nix run github:conneroisu/codex --extra-experimental-features nix-command --extra-experimental-features flakes

error: unable to execute '/nix/store/z4cwhcbj786xlw82y3mx1w3m5nxy1l4g--openai-codex-0.1.2504251709/bin/@openai/codex': No such file or directory
```

Additionally when i include this in my flake it seems to download correctly and runs without any errors however `codex`/`codex-cli` are not available in PATH. I use the `helix` flake and this seems to add to PATH correctly.

connerohnesorge committed:
fix workspace filters and proper binary path

conneroisu commented:
@peteringram0 You are correct. Additionally, it should now add to the path correctly.

connerohnesorge committed:
add a packages.default package that also point to the codex-cli package

peteringram0 commented:
> @peteringram0 You are correct. Additionally, it should now add to the path correctly. Thanks!

That update is working now! I notice though that it it exports 'codex-cli' rather than just default. This requires people to look into the flake and out what thats needed. Maybe an idea to make `.default` work?

```text
codex-cli.packages.${pkgs.system}.codex-cli
# codex-cli.packages.${pkgs.system}.default
```

connerohnesorge committed:
update depreciated devShell to devShells.default

peteringram0 commented:
@conneroisu i think in packages you can have a 'default' and just equal to 'codex-cli'. Like this: https://github.com/helix-editor/helix/blob/master/flake.nix#L50

connerohnesorge committed:
Merge branch 'main' into main

conneroisu commented:
@peteringram0 yep, didn't even see your comment, but updated to have both codex-cli, default, and the `update nix hash script`

connerohnesorge committed:
Merge branch 'main' into main

connerohnesorge committed:
added temporary sign-cla workflow

connerohnesorge committed:
missed conditional if in temporary workflow so that the bot signs the
cla

connerohnesorge committed:
remove the temporary sign workflow for the bot to sign the cla

connerohnesorge committed:
Merge branch 'main' into main

connerohnesorge committed:
Merge branch 'main' into main

actions-user committed:
chore: Update pnpm deps hash in flake.nix
This automated commit updates the pnpm deps hash hash in flake.nix to match
the latest hash of the workspace's pnpm-lock.yaml file. This ensures that
Nix's store is consistent with the latest dependencies.

connerohnesorge committed:
Merge branch 'main' into main

connerohnesorge committed:
Merge branch 'main' into main
